inluxclient/query_test.go

```diff
@@ -0,0 +1,1083 @@
+package influxclient_test
+
+import (
+	"errors"
+	"github.com/stretchr/testify/assert"
+	"io"
+	"io/ioutil"
+	"strings"
+	"testing"
+	"time"
+
+	influxclient "github.com/influxdata/influxdb-client-go/inluxclient"
+	"github.com/stretchr/testify/require"
+)
+
+func mustParseTime(s string) time.Time {
+	t, err := time.Parse(time.RFC3339, s)
+	if err != nil {
+		panic(err)
+	}
+	return t
+}
+
+type expectedTable struct {
+	columns []influxclient.TableColumn
+	rows    [][]interface{}
+}
+
+func verifyTables(t *testing.T, csv string, tables []expectedTable) {
+	reader := strings.NewReader(csv)
+	res := influxclient.NewQueryResults(ioutil.NopCloser(reader))
+
+	for _, table := range tables {
+		require.True(t, res.NextTable(), res.Err())
+		require.Nil(t, res.Err())
+		require.Equal(t, table.columns, res.Columns())
+		for _, row := range table.rows {
+			require.True(t, res.NextRow(), res.Err())
+			require.Nil(t, res.Err())
+			require.Equal(t, row, res.Values())
+			for i, c := range table.columns {
+				require.Equal(t, row[i], res.ValueByName(c.Name))
```

Comment:
I'm not that keen on `reflect.DeepEqual` being used to compare time values - it's generally considered to be bad practice.

You might consider using `github.com/frankban/quicktest` instead of testify - it makes it trivial to do this correctly (`qt.DeepEquals` uses the [cmp package](https://pkg.go.dev/github.com/google/go-cmp@v0.5.4/cmp) which understands `Equal` methods), and the API is a lot smaller.

Reply:
Ok, I will fix comparing time values. 
Generally, `testify` seems more friendly and straightforward to use to me.